main.go

```diff
@@ -162,6 +164,50 @@ func dumpJSONFile(file string, pretty bool, schema bool) error {
 	return nil
 }
 
+func getFilesContentHash(files []string, tmp string) (string, error) {
+	sha1 := sha1.New()
+
+	for i := 0; i < len(files); i++ {
+		if files[i] == "" {
+			i += 1
+			b, err := io.ReadAll(os.Stdin)
+			if err != nil {
+				return "", err
+			}
+
+			if _, err := sha1.Write(b); err != nil {
+				return "", err
+			}
+
+			file, err := os.OpenFile(tmp, os.O_RDWR, os.ModePerm)
+			if err != nil {
+				return "", err
+			}
+
+			if _, err := file.Write(b); err != nil {
+				return "", err
+			}
+
+		} else {
+			file, err := os.Open(files[i])
+			if err != nil {
+				return "", err
+			}
+			defer file.Close()
```

Comment:
defer happens at the end of functions only, not blocks. So you can't defer you need to just call `file.close` at the end of the loop

Reply:
btw file closing doesn't matter on anywhere except Windows. If one process has a file open no one else can edit it (or read it I can't remember)